File: Csharp/LcdNumbers/Scaling.cs
using System;
using Sharpen;

namespace Org.Codecop.Lcdnumbers
{
	/// <summary>The scale factor value.</summary>
	public class Scaling
	{
		public static readonly Org.Codecop.Lcdnumbers.Scaling None = new Org.Codecop.Lcdnumbers.Scaling(1);

		public static readonly Org.Codecop.Lcdnumbers.Scaling Two = new Org.Codecop.Lcdnumbers.Scaling(2);

		private readonly int times;

		public static Org.Codecop.Lcdnumbers.Scaling Of(int t)
		{
			return new Org.Codecop.Lcdnumbers.Scaling(t);
		}

		private Scaling(int times)
		{
			if (times <= 0)
			{
				throw new ArgumentException("scaling factor must be >= 1");
			}
			this.times = times;
		}

		public virtual bool None()
		{
			return times == 1;
		}

		public virtual void Times(Runnable block)
		{
			Objects.RequireNonNull(block);
			for (int i = 0; i < times; i++)
			{
				block.Run();
			}
		}

		public override bool Equals(object other)
		{
			if (!(other is Org.Codecop.Lcdnumbers.Scaling))
			{
				return false;
			}
			Org.Codecop.Lcdnumbers.Scaling that = (Org.Codecop.Lcdnumbers.Scaling)other;
			return this.times == that.times;
		}

		public override int GetHashCode()
		{
			return times;
		}

		public override string ToString()
		{
			return Sharpen.Extensions.ToString(times);
		}
	}
}

File: PHP/src/LcdNumbers/Scaling.php
<?php
namespace LcdNumbers;

/**
 * The scale factor value.
 */
class Scaling
{

    public static $NONE;
    public static $TWO;

    private $times;

    public static function of($t)
    {
        return new Scaling($t);
    }

    private function __construct($times)
    {
        if ($times <= 0) {
            throw new \InvalidArgumentException("scaling factor must be >= 1");
        }

        $this->times = $times;
    }

    public function none()
    {
        return $this->times == 1;
    }

    public function times($block)
    {
        for ($i = 0; $i < $this->times; $i++) {
            $block();
        }
    }

    public function __toString()
    {
        return (string) $this->times;
    }
}

Scaling::$NONE = Scaling::of(1);
Scaling::$TWO = Scaling::of(2);
